### PR TITLE
Better email triggers

### PR DESCRIPTION
**Pull request by tswfi**
* two stage welcome emails
  * after application is created: thank you, we will check your application
  * after application is approve / rejected: welcome / rejected email
* notification email to admins of new application

Fixes: #86

### CODE REVIEW

File: users/signals.py [hunk 1]
```diff
@@ -18,19 +18,23 @@
 #
 # Signal for other modules to activate user
```
Comment (vranki) [inline users/signals.py]: What does activating or deactivating mean in practice? There is no concept of active or inactive users. When is this emitted?

File: users/templates/mail/application_received.html [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  payment information below.
+</p>
+
+<p>
+  <b>Account holder</b>: {{ settings.ACCOUNT_NAME }}<br />
+  ACCOUNT_IBAN : {{ settings.ACCOUNT_IBAN }}<br />
+  ACCOUNT_BIC : {{ settings.ACCOUNT_BIC }}<br />
+  ref : {{ user.reference_number }}
+</p>
+
```
Comment (vranki) [inline users/templates/mail/application_received.html]: This should be included only if user has requested a member service which has a monthly fee. Also include the price of the service (MemberService.cost_string()).  

Actually this should really be generated dynamically from memberservices to handle all cases.

File: users/templates/mail/welcome_and_next_steps.html [hunk 3]
```diff
@@ -1,45 +1,67 @@
...
 
+            <h3>Bank account information is</h3>
+
+            <p>
+              <b>Account holder</b>: {{ settings.ACCOUNT_NAME }}<br />
+              ACCOUNT_IBAN : {{ settings.ACCOUNT_IBAN }}<br />
+              ACCOUNT_BIC : {{ settings.ACCOUNT_BIC }}<br />
+              ref : {{ user.reference_number }}
+            </p>
+
```
Comment (vranki) [inline users/templates/mail/welcome_and_next_steps.html]: Again, these should be generated dynamically from user's subscriptions.

I suggest to remove this section and just point a link to mulysa user page which already has all the details.

File: users/signals.py (hunk 1 above)
Comment (tswfi) [reply users/signals.py]: This is from the time when I used is_active from the user.

And the concept is still there (default django functionality). is_active user cannot login for example.

https://github.com/TampereHacklab/mulysa/pull/103/files/7a6263f1a4085fd484b0bb618c884ff87947e1d4#diff-98185baf554572b17e7545c526949658R200 emits those activate_user and deactivate_user signals when users is_active attribute changes.

File: users/templates/mail/application_received.html (hunk 2 above)
Comment (tswfi) [reply users/templates/mail/application_received.html]: added new section "services" to the emails, it lists the users services fedf25254022935731b8e972355c86e5676be769

Comment (tswfi) [reply users/templates/mail/application_received.html]: and the bank information is relevant even with only yearly fee

File: users/templates/mail/welcome_and_next_steps.html (hunk 3 above)
Comment (tswfi) [reply users/templates/mail/welcome_and_next_steps.html]: bank information is always relevant

File: users/templates/mail/welcome_and_next_steps.html [hunk 4]
```diff
@@ -1,45 +1,75 @@
...
+              highly recommendable to join our communication channels (Matrix,
+              Telegram or IRC and email list). Be happy to ask if you like to
+              know more!
+            </p>
+
+            <h2>Payment info</h2>
+            <p>
+              Your personal bank reference number is
+              <b>{{user.reference_number}}</b>. Remember always to use your
+              reference number.
```
Comment (vranki) [inline users/templates/mail/welcome_and_next_steps.html]: Nope, this is not true. Personal number is used _only_ for default service (= access rights) payments, no others. Actually personal reference number should be hidden if user is not subscribed to access right.

File: users/templates/mail/welcome_and_next_steps.html (hunk 3 above)
Comment (vranki) [reply users/templates/mail/welcome_and_next_steps.html]: Showing reference number may cause people to use it when they shouldn't.

File: www/templates/www/thanks.html [hunk 5]
```diff
@@ -1,7 +1,11 @@
 {% extends "www/base.html" %}
 {% load i18n %}
 {% block content %}
-<p>
-    {% trans 'Registration successful! Please check your e-mail (and spam folder) for instuctions.' %}
-</p>
+<div class="container-fluid">
+  <div class="row">
+    <div class="col-lg-6 offset-lg-3">
+        <h1>{% trans 'Registration successful! Please check your e-mail (and spam folder) for instuctions.' %}</h1>
```
Comment (vranki) [inline www/templates/www/thanks.html]: Should be application, not registration.. My bad, I think I wrote the string.

File: drfx/settings.py [hunk 6]
```diff
@@ -20,6 +20,8 @@
 
 CUSTOM_INVOICE_REFERENCE_BASE = 10000
 
+MEMBERSHIP_APPLICATION_NOTIFY_ADDRESS = "example@example.com"
```
Comment (vranki) [inline drfx/settings.py]: It is good custom to have example emails ending with .invalid, so example@example.com.invalid
This way mail systems don't even try to deliver to it.

Comment (tswfi) [reply drfx/settings.py]: example.com is a reserved domain name for this purpose rfc2606

File: www/templates/www/thanks.html (hunk 5 above)
Comment (tswfi) [reply www/templates/www/thanks.html]: https://github.com/TampereHacklab/mulysa/pull/103/commits/0367f169fa5e0d710d22c86704ee939c98dc0bd8

File: users/templates/mail/welcome_and_next_steps.html (hunk 4 above)
Comment (tswfi) [reply users/templates/mail/welcome_and_next_steps.html]: https://github.com/TampereHacklab/mulysa/pull/103/commits/51af7272cf788c964be2e2e362a4292c81d9b97c

now the reference number only shows for the default service

File: users/templates/mail/welcome_and_next_steps.html [hunk 7]
```diff
@@ -1,75 +1,56 @@
...
+  highly recommendable to join our communication channels (Matrix,
+  Telegram or IRC and email list). Be happy to ask if you like to
+  know more!
+</p>
+
+<h2>Payment info</h2>
+<p>
+  Your personal bank reference number is
+  <b>{{user.reference_number}}</b>. Remember always to use your
+  reference number.
```
Comment (vranki) [inline users/templates/mail/welcome_and_next_steps.html]: Still here.

Comment (tswfi) [reply users/templates/mail/welcome_and_next_steps.html]: doh

Comment (tswfi) [reply users/templates/mail/welcome_and_next_steps.html]: actually that whole paragraph is misleading now

**Pull request merged**